api/src/main/java/com/nicknathanjustin/streamercontracts/donations/DonationModelRepository.java

```diff
@@ -1,8 +1,21 @@
 package com.nicknathanjustin.streamercontracts.donations;
 
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
 
+import java.sql.Timestamp;
 import java.util.UUID;
 
 public interface DonationModelRepository extends CrudRepository<DonationModel, UUID> {
+
+    @Query("SELECT donationModel " +
+           "FROM DonationModel donationModel " +
+           "WHERE donationModel.donator.id = :donatorId " +
+           "AND (donationModel.contract.isCompleted = false OR donationModel.contract.expiresAt > :currentTimestamp)")
```

Comment:
I'm not sure if this is the query that we want. Every field that you're referencing here is also in the contract table. I know we're not supporting it for v1, but what if a contract has > 1 donation on it? Do we want this view to show just the donator's portion, or the sum of the whole contract?

Reply:
This is the query I was envisioning:

SELECT contract.description, contract.proposed_at, contract.expires_at, (SELECT streamer.twitch_username FROM users streamer where streamer.id = contract.streamer_id), (SELECT SUM(donation.donation_amount) FROM contract_donations donation WHERE donation.contract_id = contract.id) FROM users donator, contracts contract where donator.id = contract.proposer_id and donator.twitch_username = 'Walshy' and contract.is_completed = FALSE and NOW() < contract.expires_at;

This selects all of Walshy's open contracts. This would return the sum of the donations in the donation table though, not just the individual donations (in case Walshy's contract had > 1 donations)